src/graphql/node/control.rs

```diff
@@ -74,14 +91,924 @@ impl NodeControlMutation {
+
+    if log_options.is_empty() {
+        None
+    } else {
+        Some(log_options)
+    }
+}
+
+fn build_http_file_types(settings_draft: &NodeSettings) -> Option<Vec<String>> {
+    let condition_to_http_file_types = [
+        (settings_draft.office, "office"),
+        (settings_draft.exe, "exe"),
+        (settings_draft.pdf, "pdf"),
+        (settings_draft.html, "html"),
+        (settings_draft.txt, "txt"),
+    ];
+
+    let http_file_types = condition_to_http_file_types
+        .iter()
+        .filter_map(|(cond, value)| {
+            if *cond {
+                Some((*value).to_string())
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<String>>();
+
+    if http_file_types.is_empty() {
+        None
+    } else {
+        Some(http_file_types)
+    }
+}
+
+fn build_active_protocols(settings_draft: &NodeSettings) -> Option<Vec<String>> {
+    if settings_draft.protocols {
+        Some(
+            settings_draft
+                .protocol_list
+                .iter()
+                .filter_map(|(k, v)| if *v { Some(k.clone()) } else { None })
+                .collect::<Vec<String>>(),
+        )
+    } else {
+        None
+    }
+}
+
+fn build_active_sources(settings_draft: &NodeSettings) -> Option<Vec<String>> {
+    if settings_draft.sensors {
+        Some(
+            settings_draft
+                .sensor_list
+                .iter()
+                .filter_map(|(k, v)| if *v { Some(k.clone()) } else { None })
+                .collect::<Vec<String>>(),
+        )
+    } else {
+        None
+    }
+}
+
+fn build_reconverge_config(settings_draft: &NodeSettings) -> anyhow::Result<oinq::Config> {
+    let review_address = build_socket_address(
+        settings_draft.reconverge_review_ip,
+        settings_draft.reconverge_review_port,
+    )
+    .ok_or_else(|| anyhow::anyhow!("reconverge review address is not set"))?;
+
+    let giganto_address = build_socket_address(
+        settings_draft.reconverge_giganto_ip,
+        settings_draft.reconverge_giganto_port,
+    );
+
+    Ok(oinq::Config::Reconverge(ReconvergeConfig {
+        review_address,
+        giganto_address,
+    }))
+}
+
+fn build_socket_address(ip: Option<IpAddr>, port: Option<u16>) -> Option<SocketAddr> {
+    ip.and_then(|ip| port.map(|port| SocketAddr::new(ip, port)))
+}
+
+#[allow(clippy::struct_excessive_bools)]
+struct ModuleSpecificSettingUpdateIndicator {
+    review: bool,
+    hog: bool,
+    reconverge: bool,
+    piglet: bool,
+}
+
+impl ModuleSpecificSettingUpdateIndicator {
+    fn all_true(&self) -> bool {
+        self.review && self.hog && self.reconverge && self.piglet
+    }
+}
+
+fn okay_to_update_module_specific_settings(
+    setting_draft_value: bool,
+    config_setted_modules: &[ModuleName],
+    expected_module: ModuleName,
+) -> bool {
+    !setting_draft_value || config_setted_modules.iter().any(|x| *x == expected_module)
+}
+
+async fn update_node_data(
```

Comment:
`_data` looks somewhat odd to me. `update_node` would suffice and conform to what we have done.